frame/election-provider-multi-phase/src/helpers.rs

```diff
@@ -90,9 +91,33 @@ pub fn voter_index_fn_linear<T: Config>(
 	})
 }
 
-/// Create a function the returns the index a targets in the snapshot.
+/// Create a function the returns the index to a target in the snapshot.
 ///
-/// The returning index type is the same as the one defined in `T::CompactSolution::Target`.
+/// The returned index type is the same as the one defined in `T::CompactSolution::Target`.
+///
+/// Note: to the extent possible, the returned function should be cached and reused. Producing that
+/// function requires a `O(n log n)` data transform. Each invocation of that function completes
+/// in `O(log n)`.
+pub fn target_index_fn<T: Config>(
+	snapshot: &Vec<T::AccountId>,
+) -> Box<dyn '_ + Fn(&T::AccountId) -> Option<CompactTargetIndexOf<T>>> {
```

Comment:
Why do we use box here?

Reply:
As opposed to `impl Fn(...)`? Just because the peer functions do it that way. I suspect they were written pre `impl Trait`. I'll update them.

Reply:
https://github.com/paritytech/substrate/pull/8385/commits/58d60c31216f53190ed13baff7cd82174be76235